PancakeSwap-Bots/Set-Masterchef-Settings/src/agent.spec.ts

```diff
@@ -0,0 +1,230 @@
+import { TestTransactionEvent } from "forta-agent-tools/lib/test";
+import { createAddress } from "forta-agent-tools";
+import { provideHandleTransaction } from "./agent";
+import { BigNumber } from "ethers";
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData } from "./network";
+import abi from "./abi";
+
+const MOCK_ABI: string[] = ["function mockFunction(address mockAddress)"];
+
+const testMasterchef: string = createAddress("0x01");
+
+const DEFAULT_CONFIG: Record<number, NetworkData> = {
+  [Network.MAINNET]: {
+    masterChef: testMasterchef,
+  },
+};
+
+const setMigrator = (_migrator: string): Finding =>
+  Finding.fromObject({
+    name: "MasterChef Settings",
+    description: `setMigrator function called in MasterChef contract.`,
+    alertId: "CAKE-5-1",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "PancakeSwap",
+    metadata: {
+      _migrator: _migrator.toLowerCase(),
+    },
+  });
+
+const dev = (_devaddr: string): Finding =>
+  Finding.fromObject({
+    name: "MasterChef Settings",
+    description: `dev function called in MasterChef contract.`,
+    alertId: "CAKE-5-2",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "PancakeSwap",
+    metadata: {
+      _devaddr: _devaddr.toLowerCase(),
+    },
+  });
+
+const add = (_allocPoint: BigNumber, _lpToken: string, _withUpdate: boolean): Finding =>
+  Finding.fromObject({
+    name: "MasterChef Settings",
+    description: `add function called in MasterChef contract.`,
+    alertId: "CAKE-5-3",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "PancakeSwap",
+    metadata: {
+      _allocPoint: _allocPoint.toString(),
+      _lpToken: _lpToken.toLowerCase(),
+      _withUpdate: _withUpdate.toString(),
+    },
+  });
+
+const set = (_pid: BigNumber, _allocPoint: BigNumber, _withUpdate: boolean): Finding =>
+  Finding.fromObject({
+    name: "MasterChef Settings",
+    description: `set function called in MasterChef contract.`,
+    alertId: "CAKE-5-4",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "PancakeSwap",
+    metadata: {
+      _pid: _pid.toString(),
+      _allocPoint: _allocPoint.toString(),
+      _withUpdate: _withUpdate.toString(),
+    },
+  });
+
+const updateMultiplier = (multiplierNumber: BigNumber): Finding =>
+  Finding.fromObject({
+    name: "MasterChef Settings",
+    description: `updateMultiplier function called in MasterChef contract.`,
+    alertId: "CAKE-5-5",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    protocol: "PancakeSwap",
+    metadata: {
+      multiplierNumber: multiplierNumber.toString(),
+    },
+  });
+
+describe("Set Masterchef Settings bot test suite", () => {
+  const iface = new Interface(abi.CAKE_ABI);
+  const mockInterface = new Interface(MOCK_ABI);
+
+  const mockNetworkManager = new NetworkManager(DEFAULT_CONFIG, Network.MAINNET);
```

Comment:
```suggestion
  const mockNetworkManager = new NetworkManager(DEFAULT_CONFIG, Network.BSC);
```
similar to the comment above